nrfx_config.h

```diff
@@ -133,19 +146,23 @@
 #define NRFX_PRS_BOX_4_ENABLED 1
 #endif
 
-#ifdef CONFIG_NRFX_PWM
+#define NRFX_PWM_DT_ENABLED(idx) \
+	(DT_NODE_HAS_COMPAT(DT_NODELABEL(pwm##idx), nordic_nrf_pwm) && \
+	 DT_HAS_NODE(DT_NODELABEL(pwm##idx)))
+
+#if defined(CONFIG_NRFX_PWM) || DT_HAS_COMPAT(nordic_nrf_pwm)
 #define NRFX_PWM_ENABLED 1
 #endif
-#ifdef CONFIG_NRFX_PWM0
+#if defined(CONFIG_NRFX_PWM0) || NRFX_PWM_DT_ENABLED(0)
 #define NRFX_PWM0_ENABLED 1
```

Comment:
This comment concerns also https://github.com/zephyrproject-rtos/zephyr/pull/24473, which is tightly coupled with this PR.

I am not convinced that this is the right approach. What if one wants to use a different driver than `nrfx_pwm.c` for handling the PWM peripheral, or in general wants to use it in a custom way and only wants the information from DT (hence enables the node) but does not want the nrfx_pwm driver to be compiled?

Currently, it is possible to use a given nRF PWM peripheral in Zephyr in three ways:
- use only its DT node properties, without automatically activating any PWM driver
- use it through the Zephyr PWM API, after setting `CONFIG_PWM=y` and `CONFIG_PWM_x=y` (and as I understand we want to eliminate the need of setting the latter); this currently uses `pwm_nrfx.c` and `nrfx_pwm.c` underneath for for all instances, but it could be potentially extended with some other implementation of the driver
- use it through the nrfx_pwm API, after setting `CONFIG_NRFX_PWMx=y` while `CONFIG_PWM=n` (`pwm_nrfx.c` will not be compiled then, only `nrfx_pwm.c`)

I think we should keep all these possibilities.